docs/howto/garbage-collection.md

```diff
@@ -0,0 +1,254 @@
+
+To define retention rules, either use the `lakectl` command or the lakeFS web UI:
+
+<div class="tabs">
+  <ul>
+    <li><a href="#lakectl-option">CLI</a></li>
+    <li><a href="#ui-option">Web UI</a></li>
+  </ul>
+  <div markdown="1" id="lakectl-option">
+
+Create a JSON file with your GC rules:
+
+```bash
+cat <<EOT >> example_repo_gc_rules.json
+{
+  "default_retention_days": 14,
+  "branches": [
+    {"branch_id": "main", "retention_days": 21},
+    {"branch_id": "dev", "retention_days": 7}
+  ]
+}
+EOT
+```
+
+Set the GC rules using `lakectl`:
+```bash
+lakectl gc set-config lakefs://example-repo -f example_repo_gc_rules.json 
+```
+
+</div>
+<div markdown="1" id="ui-option">
+From the lakeFS web UI:
+
+1. Navigate to the main page of your repository.
+2. Go to _Settings_ -> _Retention_.
+3. Click _Edit policy_ and paste your GC rule into the text box as a JSON.
+4. Save your changes.
+
+![GC Rules From UI]({{ site.baseurl }}/assets/img/gc_rules_from_ui.png)
+</div>
+</div>
+
+## Running the GC job
+
+To run the job, use the following `spark-submit` command (or using your preferred method of running Spark programs).
+
+<div class="tabs">
+  <ul>
+    <li><a href="#aws-option">On AWS (Spark 3.1.2 and higher)</a></li>
+    <li><a href="#aws-301-option">On AWS (Spark 3.0.1)</a></li>
+    <li><a href="#azure-option">On Azure</a></li>
+    <li><a href="#gcp-option">On GCP</a></li>
+  </ul>
+  <div markdown="1" id="aws-option">
+  ```bash
+spark-submit --class io.treeverse.gc.GarbageCollection \
+  --packages org.apache.hadoop:hadoop-aws:2.7.7 \
+  -c spark.hadoop.lakefs.api.url=https://lakefs.example.com:8000/api/v1  \
+  -c spark.hadoop.lakefs.api.access_key=<LAKEFS_ACCESS_KEY> \
+  -c spark.hadoop.lakefs.api.secret_key=<LAKEFS_SECRET_KEY> \
+  -c spark.hadoop.fs.s3a.access.key=<S3_ACCESS_KEY> \
+  -c spark.hadoop.fs.s3a.secret.key=<S3_SECRET_KEY> \
+  http://treeverse-clients-us-east.s3-website-us-east-1.amazonaws.com/lakefs-spark-client-312-hadoop3/0.8.1/lakefs-spark-client-312-hadoop3-assembly-0.8.1.jar \
+  example-repo us-east-1
+  ```
+  </div>
+  <div markdown="1" id="aws-301-option">
+  ```bash
+spark-submit --class io.treeverse.gc.GarbageCollection \
+  --packages org.apache.hadoop:hadoop-aws:2.7.7 \
+  -c spark.hadoop.lakefs.api.url=https://lakefs.example.com:8000/api/v1  \
+  -c spark.hadoop.lakefs.api.access_key=<LAKEFS_ACCESS_KEY> \
+  -c spark.hadoop.lakefs.api.secret_key=<LAKEFS_SECRET_KEY> \
+  -c spark.hadoop.fs.s3a.access.key=<S3_ACCESS_KEY> \
+  -c spark.hadoop.fs.s3a.secret.key=<S3_SECRET_KEY> \
+  http://treeverse-clients-us-east.s3-website-us-east-1.amazonaws.com/lakefs-spark-client-301/0.8.1/lakefs-spark-client-301-assembly-0.8.1.jar \
+  example-repo us-east-1
+  ```
+  </div>
+
+  <div markdown="1" id="azure-option">
+
+If you want to access your storage using the account key:
+
+  ```bash
+spark-submit --class io.treeverse.gc.GarbageCollection \
+  --packages org.apache.hadoop:hadoop-aws:3.2.1 \
```

Comment:
```suggestion
  --packages org.apache.hadoop:hadoop-azure:3.2.1 \
```
no?

Reply:
Is `hadoop-aws` needed here?
I know it's not your addition, yet it might be redundant...